views/NewChart.react.js

```diff
@@ -1,9 +1,103 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react'
+import { compose } from 'redux';
+import { withStyles } from '@material-ui/core/styles';
+import Button from '@material-ui/core/Button'
+import { connect } from 'react-redux';
+import Header from './components/Header';
+import Sidebar from './components/Sidebar';
+import BarChartFields from './forms/BarChartFields'
+import getCounts from './fe-utils/countUtil';
+import getAvatar from './fe-utils/avatarUtil';
+import { fetchSubjects, createChart } from './fe-utils/fetchUtil';
+import getDefaultStyles from './fe-utils/styleUtil';
+import { chartStyles } from './styles/chart_styles';
+
+const NewChart = ({ user, classes }) => {
+  const [formValues, setFormValues] = useState({})
+  const [valueAndLabelFields, setValueAndLabelFields] = useState([])
+  const [openDrawer, setOpenDrawer] = useState(false)
+  const [sideBarState, setSideBarState] = useState({ totalDays: 0, totalStudies: 0, totalSubjects: 0 })
+  const [avatar, setAvatar] = useState('');
+
+  const toggleDrawer = () => setOpenDrawer(!openDrawer)
+  const updateFormValues = (e,) => setFormValues({...formValues, [e.target.name]: e.target.value})
+  const addValueAndLabelField = () => setValueAndLabelFields(prevState => [{ value: '', label: '' }, ...prevState ])
+  const removeValueAndLabelField = (id) => setValueAndLabelFields(prevState => [...prevState.filter((_, index) => index !== id)])
+  const handleValueAndLabelFieldUpdate = (e, id) => setValueAndLabelFields(prevState => 
+    [...prevState
+      .map((field, idx) => 
+        id === idx
+        ? 
+          ({ ...field, [e.target.name]: e.target.value }) 
+        : 
+          ({...field})
+        )
+      ])
+  const handleSubmit = async (e) => {
+    e.preventDefault()
+    const values = { ...formValues, fields: valueAndLabelFields, owner: user.uid }
+    await createChart(values)
+  }
+
+  useEffect(() => {
+    fetchSubjects().then(acl => {
+      setSideBarState(getCounts({ acl }))
+    })
+    setAvatar(getAvatar({ user: user }))
+  }, [])
 
-export default function NewChart() {
   return (
-    <div>
-      <h1> New Chart</h1>
+    <div className={ classes.root }>
+      <Header
+        handleDrawerToggle={ toggleDrawer }
+        title={'Create a New Chart'}
+        isAccountPage={false}
+      />
+      <Sidebar
+        avatar={avatar}
+        handleDrawerToggle={ toggleDrawer }
+        mobileOpen={ openDrawer }
+        totalDays={ sideBarState.totalDays }
+        totalStudies={ sideBarState.totalStudies }
+        totalSubjects={ sideBarState.totalSubjects }
+        user={user}
+      />
+      <>
+        <div className={`${ classes.content } ${ classes.contentPadded }`}>
+          <form autoComplete='off' onSubmit={ handleSubmit }>
```

Comment:
can we extract this form to a new component that receives a `handleSubmit` prop? I think forms are a nice separation of concerns for these larger view components so that they don't get too large and cumbersome.

Reply:
[fd611f7](https://github.com/AMP-SCZ/dpdash/pull/84/commits/fd611f7e8aaef083e1cbd59a407f40a134bd1863)

Reply:
I don't think I was being clear earlier, sorry for the confusion. I meant the entire chart form, not just the `form` html element/jsx component.